appstore/validator.go

```diff
@@ -170,7 +170,11 @@ func (c *Client) parseResponse(resp *http.Response, result interface{}, ctx cont
 			return fmt.Errorf("Received http status code %d from the App Store Sandbox: %w", resp.StatusCode, ErrAppStoreServer)
 		}
 
-		return json.NewDecoder(resp.Body).Decode(result)
+		err = json.NewDecoder(resp.Body).Decode(result)
+		if err != nil {
+			return err
+		}
+		return ErrReceiptIsForTest
```

Comment:
If it was a receipt for the correct Sandbox, the request to the Sandbox should succeed, and if all requests are treated as errors, there seems to be a problem 👀 

>Verify your receipt first with the production URL; then verify with the sandbox URL if you receive a 21007 status code. This approach ensures you don’t have to switch between URLs while your application is tested, reviewed by App Review, or live in the App Store.
>
>https://developer.apple.com/documentation/storekit/in-app_purchase/original_api_for_in-app_purchase/validating_receipts_with_the_app_store

Reply:
@takecy, Thank you very much for your review. Yes, you are right, when the receipt is the correct Sandbox receipt, the `ErrReceiptIsForTest` could be returned rather than nil. 

Another option could return the status code directly in the function `Verify`. Could you please review another PR https://github.com/awa/go-iap/pull/195/commits/5bd8c8fea386bb8b9816b81d0a05c51bdaf7eff7.

Reply:
@richzw 
Thanks for another option :) looks good 👍 
However, since the `Verify` function is already exposed, I think changing the signature would break backward compatibility.
How about adding a new function with this handling (e.g. `VerifyWithStatus` ) to the interface?

Reply:
@takecy , thank you very much for your quick review. The `VerifyWithStatus` has been added in this PR https://github.com/awa/go-iap/pull/195/commits/6fa5b566de3ef01c28366dd37a7ffe8ef540f730. Could you please review it again?

Reply:
@richzw 
Looks very good 😁 
There seems to be a compile error in the test code, so please just fix that.

Reply:
@takecy , Sorry for the missing test code error. Just fix it.